qemu/tests/disk_extension.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import time
+
+from avocado.utils import process
+from virttest import env_process
+from virttest import error_context, data_dir
+from virttest.qemu_storage import QemuImg
+from virttest.utils_misc import get_linux_drive_path
+
+
+# This decorator makes the test function aware of context strings
+@error_context.context_aware
+def run(test, params, env):
+    """
+    QEMU 'disk images extension in io-error status' test
+
+    1) Create folder and mounted it as tmpfs type.
+    2) Create a raw image file with small size(10M) under the tmpfs folder.
+    3) Attach loop device with above raw image file.
+    4) Create qcow2 image on the loop device with larger size (500M).
+    5) Boot vm with loop device as data disk.
+    6) Access  guest vm and execute dd operation on the data disk.
+     the IO size is same as the loop device virtual disk size.
+    7) Verify vm status is io-error status in hmp.
+    8) Continue to increase disk size of the raw image file and update the loop device.
+    9) Verify vm status whether in expected status:
+      if the raw image file size is smaller than loop device virtual disk size,
+      it is in io-error status,Otherwise it is in running status.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def prepare_tmpfs_folder(dirname):
+        logging.debug("Ready to create tmpfs folder on %s" % dirname)
+        cmd = "umount -l {0};rm -rf {0};mkdir -p {0};".format(dirname)
+        process.system_output(cmd, ignore_status=True, shell=True)
+        cmd = "mount -t tmpfs -o rw,nosuid,nodev,seclabel tmpfs {}".format(dirname)
+        process.system_output(cmd, shell=True)
+
+    def setup_loop_device_on_image(backend_img, device_img):
+        if backend_img.image_format != "raw":
+            test.error("Only support attaching loop device on raw format image.")
+            return
+        backend_img.create(backend_img.params)
+        backend_filename = backend_img.image_filename
+        loop_device_name = device_img.image_filename
+        logging.debug("Ready to create loop device image {} on {}".format(loop_device_name, backend_filename))
+        cmd = "losetup -d {}".format(loop_device_name)
+        process.system_output(cmd, ignore_status=True, shell=True)
```

Comment:
Why ignore the status of command? will not block the next steps if fail to execute the command?

Reply:
if we want execute test in new ENV, we do not care the loop device was attached before or not. it is not block next step